learning_resources/serializers.py

```diff
@@ -103,29 +134,202 @@ class LearningResourceBaseSerializer(serializers.ModelSerializer):
+            missing = set(topics).difference(valid_topic_ids)
+            if missing:
+                raise ValidationError(f"Invalid topic ids: {missing}")
+        return {"topics": topics}
+
+    @extend_schema_field(LearningResourceTopicSerializer(many=True, allow_null=True))
+    def get_topics(self, instance):
+        """Returns the list of topics"""
+        return [
+            LearningResourceTopicSerializer(topic).data
+            for topic in instance.topics.all()
+        ]
 
     class Meta:
         model = models.LearningResource
-        fields = "__all__"
+        exclude = ["resources", *COMMON_IGNORED_FIELDS]
 
 
 class ProgramSerializer(serializers.ModelSerializer):
     """Serializer for the Program model"""
 
-    courses = LearningResourceBaseSerializer(read_only=True, many=True, allow_null=True)
+    courses = serializers.SerializerMethodField()
+
+    @extend_schema_field(LearningResourceBaseSerializer(many=True, allow_null=True))
+    def get_courses(self, obj):
+        """Get the learning resource courses for a program"""
+        return LearningResourceRelationshipChildField(
+            obj.learning_resource.children.all(), many=True
+        ).data
 
     class Meta:
         model = models.Program
         exclude = ("learning_resource", *COMMON_IGNORED_FIELDS)
 
 
 class LearningResourceSerializer(LearningResourceBaseSerializer):
-    """Full serializer for LearningResource"""
+    """Serializer for LearningResource, with program included"""
 
-    course = CourseSerializer(read_only=True, allow_null=True)
     program = ProgramSerializer(read_only=True, allow_null=True)
-    runs = LearningResourceRunSerializer(read_only=True, many=True, allow_null=True)
+
+
+class LearningResourceRelationshipChildField(serializers.ModelSerializer):
+    """
+    Serializer field for the LearningResourceRelationship model that uses
+    the LearningResourceSerializer to serialize the child resources
+    """
+
+    def to_representation(self, instance):
+        """Serializes child as a LearningResource"""
+        return LearningResourceSerializer(instance=instance.child).data
+
+    class Meta:
+        model = models.LearningResourceRelationship
+        exclude = ("parent", *COMMON_IGNORED_FIELDS)
+
+
+class LearningPathResourceSerializer(LearningResourceSerializer):
+    """CRUD serializer for LearningPath resources"""
+
+    def validate_resource_type(self, value):
+        """Only allow LearningPath resources to be CRUDed"""
+        if value != LearningResourceType.learning_path.value:
+            raise serializers.ValidationError(
+                "Only LearningPath resources are editable"
+            )
+        return value
+
+    def create(self, validated_data):
+        """Ensure that the LearningPath is created by the requesting user; set topics"""
+        request = self.context.get("request")
+        if request and hasattr(request, "user") and isinstance(request.user, User):
```

Comment:
Are we able to move this kind of validation into the view? To create a learning path we already should have an authenticated user.

Reply:
Yes, user shouldn't even get this far if not authenticated so conditional can just be removed